packages/@aws-cdk/aws-appmesh/lib/tls-validation.ts

```diff
@@ -0,0 +1,196 @@
+   */
+  readonly certificateChain: string;
+}
+
+/**
+ * Defines the TLS validation context trust.
+ */
+export abstract class TlsValidationTrust {
+  /**
+   * Tells envoy where to fetch the validation context from
+   */
+  public static file(props: TlsValidationFileTrustOptions): TlsValidationTrust {
+    return new TlsValidationFileTrustImpl(props);
+  }
+
+  /**
+   * TLS validation context trust for ACM Private Certificate Authority (CA).
+   */
+  public static acm(props: TlsValidationAcmTrustOptions): TlsValidationTrust {
+    return new TlsValidationAcmTrustImpl(props);
+  }
+
+  /**
+   * Returns Trust context based on trust type.
+   */
+  public abstract bind(scope: Construct): TlsValidationTrustConfig;
+
+}
+
+class TlsValidationAcmTrustImpl extends TlsValidationTrust {
+  /**
+   * Contains information for your private certificate authority
+   */
+  readonly certificateAuthorities: acmpca.ICertificateAuthority[];
+
+  constructor (props: TlsValidationAcmTrustOptions) {
+    super();
+    this.certificateAuthorities = props.certificateAuthorities;
+  }
+
+  public bind(_scope: Construct): TlsValidationTrustConfig {
+    if (this.certificateAuthorities.length === 0) {
+      throw new Error('you must provide at least one Certificate Authority when creating an ACM Trust ClientPolicy');
+    } else {
+      return {
+        virtualNodeClientTlsValidationTrust: {
+          acm: {
+            certificateAuthorityArns: this.certificateAuthorities.map(certificateArn =>
+              certificateArn.certificateAuthorityArn),
+          },
+        },
+        virtualGatewayClientTlsValidationTrust: {
+          acm: {
+            certificateAuthorityArns: this.certificateAuthorities.map(certificateArn =>
+              certificateArn.certificateAuthorityArn),
+          },
+        },
+      };
+    }
+  }
+}
+
+class TlsValidationFileTrustImpl extends TlsValidationTrust {
+  /**
+   * Path to the Certificate Chain file on the file system where the Envoy is deployed.
+   */
+  readonly certificateChain: string;
+
+  constructor (props: TlsValidationFileTrustOptions) {
+    super();
+    this.certificateChain = props.certificateChain;
+  }
+
+  public bind(_scope: Construct): TlsValidationTrustConfig {
+    return {
+      virtualNodeClientTlsValidationTrust: {
+        file: {
+          certificateChain: this.certificateChain,
+        },
+      },
+      virtualNodeListenerTlsValidationTrust: {
+        file: {
+          certificateChain: this.certificateChain,
+        },
+      },
+      virtualGatewayClientTlsValidationTrust: {
+        file: {
+          certificateChain: this.certificateChain,
+        },
+      },
+      virtualGatewayListenerTlsValidationTrust: {
+        file: {
+          certificateChain: this.certificateChain,
+        },
+      },
+    };
+  }
+}
+
+export function renderTlsPolicy(scope: Construct, tlsClientPolicy: TlsClientPolicy,
```

Comment:
Why is this exported from this file? This makes it part of the public API of the AppMesh module, which I'm pretty sure we don't want.

If you need this in multiple files, then you need to move this function to a separate file, preferably in the `lib/private` subdirectory, so it's obvious it's part of the public API. There's a file called `utils.ts` there, maybe this is a good place for this function.

Reply:
You are right that we do not want to make this function public. I was not sure where to put and I will follow your instruction. Thank you!